cloud_automation/java/dataflow/src/main/java/org/datacommons/dataflow/CsvImport.java

```diff
@@ -0,0 +1,121 @@
+          headerBytes[i] = Bytes.toBytes(headers[i]);
+          setCell =  Mutation.SetCell.newBuilder()
+                          .setFamilyName(FAMILY)
+                          .setColumnQualifier(ByteString.copyFrom( headerBytes[i]))
+                          .setValue(ByteString.copyFrom(Bytes.toBytes(values[i])))
+                          .build();
+           mutations.add(Mutation.newBuilder().setSetCell(setCell).build());
+        }
+        Iterable<Mutation> mutIter = mutations.build();
+        c.output(KV.of(ByteString.copyFrom(rowkey), mutIter));
+      } catch (Exception e) {
+        LOG.error("Failed to process input {}", c.element(), e);
+          throw e;
+      }
+    }
+  };
+
+  public static interface BigtableCsvOptions extends CloudBigtableOptions {
+
+    @Description("The headers for the CSV file.")
+    ValueProvider<String> getHeaders();
+
+    @SuppressWarnings("unused")
+    void setHeaders(ValueProvider<String> headers);
+
+    @Description("The Cloud Storage path to the CSV file.")
+    ValueProvider<String> getInputFile();
+
+    @SuppressWarnings("unused")
+    void setInputFile(ValueProvider<String> location);
+  }
+
+
+  /**
+   * <p>Creates a dataflow templae with following stages: </p>
+   * <ol>
+   * <li> Read a CSV from GCS.
+   * <li> Transform each line to a Bigtable Row mutation.
+   * <li> Write to Bigtable.
+   * </ol>
+   *
+   * @param args Arguments to use to configure the Dataflow Pipeline. Check README for deployment
+   * instructions.
+   * --project=[dataflow project] \\ --stagingLocation=gs://[your google storage bucket] \\
+   * --inputFile=gs://[your google storage object] \\
+   * --bigtableProject=[bigtable project] \\ --bigtableInstanceId=[bigtable instance id] \\
+   * --bigtableTableId=[bigtable tableName]
+   *
+   *
+   *  This pipepline uses BigtableIO instead of CloudBigtableIO libraries. This is because,
+   *  CloudBigtableIO libraries do not completely support dataflow templates.
+   *  The inspiration for this is pipeline is from cloud google ataflow examples.
+   */
+
+  public static void main(String[] args) throws IllegalArgumentException {
+    BigtableCsvOptions options =
+        PipelineOptionsFactory.fromArgs(args).withValidation().as(BigtableCsvOptions.class);
+
+    Pipeline p = Pipeline.create(options);
+
+    BigtableIO.Write write = BigtableIO.write()
+            .withProjectId(options.getBigtableProjectId())
+            .withInstanceId(options.getBigtableInstanceId())
+            .withTableId(options.getBigtableTableId());
+    p.apply("ReadMyFile", TextIO.read().from(options.getInputFile()))
+        .apply("TransformParsingsToBigtable", ParDo.of(new CsvToBigtableFn()))
+        .apply("WriteToBigtable", write);
+
+    PipelineResult result = p.run();
+            // Wait for pipeline to finish only if it is not constructing a template.
+    if (options.as(DataflowPipelineOptions.class).getTemplateLocation() == null) {
```

Comment:
For my understanding, what will poll the job running when this is fired off as template? (Sorry if you have this in a doc, I can read that too.)

Reply:
I wrote a airflow job and this template runs as part of the airflow job.
once the template(dataflow) jo completes, airflow writes to GCS where it suceeded or failed. A GCF function then triggers of this file and scales down bigtable.
Any more reports/emails we need can be triggered.

Reply:
Thanks. Might be worth documenting the final flow in the doc you had or in a README.

Reply:
yes I will update the document and readme as well.
Sorry about this going back-and-forth. Airflow is the cleanest way to report back. I thought of polling as well.